x-pack/plugins/apm/server/lib/service_map/get_max_anomalies.ts

```diff
@@ -0,0 +1,188 @@
+  }
+
+  let mlJobIds: string[] = [];
+  try {
+    mlJobIds = await getMLJobIds(ml, environment);
+  } catch (error) {
+    logger.error(error);
+    return DEFAULT_VALUE;
+  }
+
+  const params = {
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              term: {
+                result_type: 'record',
+              },
+            },
+            {
+              terms: {
+                job_id: mlJobIds,
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: start,
+                  lte: end,
+                  format: 'epoch_millis',
+                },
+              },
+            },
+            {
+              terms: {
+                by_field_value: [TRANSACTION_REQUEST, TRANSACTION_PAGE_LOAD],
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        service_name: {
+          terms: {
+            field: 'partition_field_value',
+          },
+          aggs: {
+            top_score: {
+              top_hits: {
+                sort: {
+                  record_score: 'desc',
+                },
+                _source: {
+                  includes: ['actual', 'job_id', 'by_field_value'],
+                },
+                size: 1,
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+  const response = await ml.mlSystem.mlAnomalySearch(params);
+  return {
+    mlJobIds,
+    maxAnomalies: transformResponseToMaxAnomalies(
+      response as MaxAnomaliesAggResponse
+    ),
+  };
+}
+
+interface MaxAnomaliesAggResponse {
+  aggregations: {
+    service_name: {
+      buckets: Array<{
+        key: string;
+        top_score: {
+          hits: {
+            hits: Array<{
+              sort: [number];
+              _source: {
+                actual: [number];
+                job_id: string;
+                by_field_value: string;
+              };
+            }>;
+          };
+        };
+      }>;
+    };
+  };
+}
+
+function transformResponseToMaxAnomalies(
+  response: MaxAnomaliesAggResponse
+): Array<{
+  [SERVICE_NAME]: string;
+  maxAnomaly: MaxAnomaly;
+}> {
+  const services = response.aggregations.service_name.buckets.map(
+    ({ key: serviceName, top_score: topScoreAgg }) => {
+      return {
+        [SERVICE_NAME]: serviceName,
+        maxAnomaly: {
+          [TRANSACTION_TYPE]: topScoreAgg.hits.hits[0]?._source?.by_field_value,
+          anomaly_score: topScoreAgg.hits.hits[0]?.sort?.[0],
+          actual_value: topScoreAgg.hits.hits[0]?._source?.actual?.[0],
+          job_id: topScoreAgg.hits.hits[0]?._source?.job_id,
+        },
+      };
+    }
+  );
+  return services;
+}
+
+export async function getMLJobIds(
+  ml: Required<Setup>['ml'],
+  environment?: string
+) {
+  const response = await ml.anomalyDetectors.jobs(ML_GROUP_NAME_APM);
+  const apmAnomalyDetectionMLJobs = response.jobs.filter(
```

Comment:
What about a shorter name? :)

```suggestion
  const mlJobs = response.jobs.filter(
```